test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,266 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	siteEditorStyleVariations: async (
+		{ admin, editor, page, pageUtils, requestUtils },
+		use
+	) => {
+		await use(
+			new SiteEditorStyleVariations( {
+				admin,
+				editor,
+				page,
+				pageUtils,
+				requestUtils,
+			} )
+		);
+	},
+} );
+
+test.describe( 'Global styles variations', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme(
+			'gutenberg-test-themes/style-variations'
+		);
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'should have three variations available with the first one being active', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await page.click( 'role=button[name="Styles"i]' );
+		await siteEditorStyleVariations.openOtherStyles();
+
+		const variations = page.locator(
+			'.edit-site-global-styles-variations_item'
```

Comment:
Let's not depend on class names in our selectors. Other than fixing the `aria-label` and `aria-current` of the buttons, maybe we should also group them under an accessible section too.

<img width="413" alt="image" src="https://user-images.githubusercontent.com/7753001/173989643-3b6229b2-d692-4cca-b8e9-bd5244c2f341.png">

We already have an `h5` and a `p` in the styles region, but they are not associated with any section or with each other. Maybe we could group them in a `<section>` or something else and add `aria-labelledby` and `aria-describedby` etc to provide clear accessible information?

It sounds like a bigger change than this PR though, so I agree that we can defer this to another PR and just an explanation comment above here for now.